### PR TITLE
Fix & Improve Legacy Admin Asset Loading

### PR DESCRIPTION
**Pull request by tn3rb**
see: https://github.com/eventespresso/event-espresso-core/issues/2971#issuecomment-651284186

This PR fixes the above issue as well as some other missing asset dependencies that were occurring throughout the EE admin and also moves some of the asset handling into appropriate `AssetManager` classes for better SRP

### CODE REVIEW

File: core/admin/EE_Admin_Page.core.php [hunk 1]
```diff
@@ -1887,141 +1889,32 @@ private function _add_global_feature_pointers()
...
         if (WP_DEBUG) {
             add_action('admin_head', array($this, 'add_xdebug_style'));
         }
-        // register all styles
-        wp_register_style(
-            'espresso-ui-theme',
-            EE_GLOBAL_ASSETS_URL . 'css/espresso-ui-theme/jquery-ui-1.10.3.custom.min.css',
-            array(),
-            EVENT_ESPRESSO_VERSION
-        );
```
Comment (tn3rb) [inline core/admin/EE_Admin_Page.core.php:1901]: extracted asset registration from `core/admin/EE_Admin_Page.core.php` and moved into `EspressoLegacyAdminAssetManager` and `LegacyAccountingAssetManager` for better SRP

File: core/domain/entities/routing/handlers/admin/EspressoEventEditor.php [hunk 2]
```diff
@@ -27,12 +27,8 @@ class EspressoEventEditor extends EspressoEventsAdmin
      */
     public function matchesCurrentRequest()
     {
-        global $pagenow;
         return parent::matchesCurrentRequest()
-               && $pagenow
-               && $pagenow === 'admin.php'
                && $this->admin_config->useAdvancedEditor()
-               && $this->request->getRequestParam('page') === 'espresso_events'
```
Comment (tn3rb) [inline core/domain/entities/routing/handlers/admin/EspressoEventEditor.php:35]: the removed code was already happening in `parent::matchesCurrentRequest()`

File: core/domain/services/assets/CoreAssetManager.php [hunk 3]
```diff
@@ -31,12 +31,6 @@ class CoreAssetManager extends AssetManager
 {
 
     // WordPress core / Third party JS asset handles
-    const JS_HANDLE_JQUERY = 'jquery';
-
-    const JS_HANDLE_JQUERY_VALIDATE = 'jquery-validate';
-
-    const JS_HANDLE_JQUERY_VALIDATE_EXTRA = 'jquery-validate-extra-methods';
-
```
Comment (tn3rb) [inline core/domain/services/assets/CoreAssetManager.php:39]: this and other jQuery related logic extracted to `JqueryAssetManager`

File: core/domain/services/assets/EspressoLegacyAdminAssetManager.php [hunk 4]
```diff
@@ -46,48 +78,140 @@ public function addAssets()
...
+                JqueryAssetManager::JS_HANDLE_JQUERY_UI_TIMEPICKER_ADDON,
+                EspressoLegacyAdminAssetManager::JS_HANDLE_MOMENT,
+            ]
+        );
+
+        // google charts
+        $this->addJavascript(
+            EspressoLegacyAdminAssetManager::JS_HANDLE_GOOGLE_CHARTS,
+            'https://www.gstatic.com/charts/loader.js'
+        );
```
Comment (manzoorwanijk) [inline core/domain/services/assets/EspressoLegacyAdminAssetManager.php:168]: A question not related to this PR. Do we add[ Subresource Integrity](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity) for external scripts?

File: core/domain/services/assets/JqueryAssetManager.php [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        $this->addJavascript(
+            JqueryAssetManager::JS_HANDLE_JQUERY_UI_TIMEPICKER_ADDON,
+            EE_GLOBAL_ASSETS_URL . 'scripts/jquery-ui-timepicker-addon.js',
+            [
+                JqueryAssetManager::JS_HANDLE_JQUERY_UI_DATEPICKER,
+                JqueryAssetManager::JS_HANDLE_JQUERY_UI_SLIDER
+            ]
+        );
+	}
+}
```
Comment (manzoorwanijk) [inline core/domain/services/assets/JqueryAssetManager.php:68]: Missing EOL

File: core/domain/services/assets/EspressoLegacyAdminAssetManager.php (hunk 4 above)
Comment (tn3rb) [reply core/domain/services/assets/EspressoLegacyAdminAssetManager.php:168]: oooo no we do not, that would be a nice enhancement though

File: core/domain/services/assets/JqueryAssetManager.php (hunk 5 above)
Comment (tn3rb) [reply core/domain/services/assets/JqueryAssetManager.php:68]: hmmm... wonder why linting passed ?

good catch thnx

Comment (manzoorwanijk) [reply core/domain/services/assets/JqueryAssetManager.php:68]: Yes, should work. 👍

**Pull request merged**